src/picklescan/scanner.py

```diff
@@ -307,12 +324,37 @@ def scan_pickle_bytes(data: IO[bytes], file_id, multiple_pickles=True) -> ScanRe
     return _build_scan_result_from_raw_globals(raw_globals, file_id)
 
 
+# XXX: it appears there is not way to get the byte stream for a given file within the 7z archive and thus forcing us to unzip to disk before scanning
+def scan_7z_bytes(data: IO[bytes], file_id) -> ScanResult:
+    try:
+        import py7zr
+    except ImportError:
+        raise Exception(
+            "py7zr is required to scan 7z archives, install picklescan using: 'pip install picklescan[7z]'"
+        )
+    result = ScanResult([])
+
+    with py7zr.SevenZipFile(data, mode="r") as archive:
+        file_names = archive.getnames()
+        targets = [f for f in file_names if f.endswith(tuple(_pickle_file_extensions))]
+        _log.debug("Files in 7z archive %s: %s", file_id, targets)
+        with TemporaryDirectory() as tmpdir:
+            archive.extract(path=tmpdir, targets=targets)
```

Comment:
I'm assuming extract here sanitises the output @mmaitre314

Reply:
That's my understanding. I saw a path-traversal CVE on py7zr but it has been fixed in the package version used here. https://github.com/advisories/GHSA-m8xw-9x5x-6vh3